### PR TITLE
Issue/290 list with creation

### PR DESCRIPTION
**Pull request by RomuDeuxfois**
Front end part for the XLS chronogram creation

### CODE REVIEW

File: Add_'update'_column_with_sort.patch [hunk 1]
```diff
@@ -0,0 +1,175 @@
+Subject: [PATCH] Add 'update' column with sort
```
Comment (Dimfacion) [inline Add_'update'_column_with_sort.patch:1]: This shouldn't be here

File: openbas-api/src/main/java/io/openbas/rest/mapper/form/MapperAddInput.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import lombok.Setter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.openbas.config.AppConfig.MANDATORY_MESSAGE;
+
+@Getter
+@Setter
+public class MapperAddInput {
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/rest/mapper/form/MapperAddInput.java:17]: I think we need to remove it right @Dimfacion ?

File: openbas-api/src/main/java/io/openbas/rest/mapper/form/MapperUpdateInput.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Pattern;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.openbas.config.AppConfig.MANDATORY_MESSAGE;
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/rest/mapper/form/MapperUpdateInput.java:13]: I think we need to remove it right @Dimfacion ?

File: openbas-front/package.json [hunk 4]
```diff
@@ -60,6 +60,7 @@
     "react-markdown": "9.0.1",
     "react-redux": "8.1.3",
     "react-router-dom": "6.23.1",
+    "react-syntax-highlighter": "^15.5.0",
```
Comment (RomuDeuxfois) [inline openbas-front/package.json:63]:
```suggestion
    "react-syntax-highlighter": "15.5.0",
```

File: openbas-front/package.json [hunk 5]
```diff
@@ -84,6 +85,7 @@
     "@types/react": "18.3.3",
     "@types/react-csv": "1.1.10",
     "@types/react-dom": "18.3.0",
+    "@types/react-syntax-highlighter": "^15",
```
Comment (RomuDeuxfois) [inline openbas-front/package.json:88]:
```suggestion
    "@types/react-syntax-highlighter": "15",
```

File: openbas-front/src/actions/xls_formatter/xls-formatter-actions.ts [hunk 6]
```diff
@@ -1,5 +1,5 @@
-import type { RawPaginationImportMapper, SearchPaginationInput } from '../../utils/api-types';
-import { simpleDelCall, simplePostCall } from '../../utils/Action';
+import type { ImportMapperAddInput, ImportMapperUpdateInput, InjectsImportTestInput, RawPaginationImportMapper, SearchPaginationInput } from '../../utils/api-types';
```
Comment (RomuDeuxfois) [inline openbas-front/src/actions/xls_formatter/xls-formatter-actions.ts:1]: The file name should be xlsMapper-action to match the MapperApi

File: openbas-front/src/actions/xls_formatter/xls-formatter-actions.ts [hunk 7]
```diff
@@ -9,7 +9,33 @@ export const searchMappers = (searchPaginationInput: SearchPaginationInput) => {
   return simplePostCall(uri, data);
 };
 
+export const fetchMapper = (mapperId: string) => {
```
Comment (RomuDeuxfois) [inline openbas-front/src/actions/xls_formatter/xls-formatter-actions.ts:12]: fetchXlsMapper

File: openbas-front/src/admin/components/common/injects/ImportUploaderInject.tsx [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+import React, { useContext, useState } from 'react';
+import Dialog from '../../../../components/common/Dialog';
+import { useFormatter } from '../../../../components/i18n';
+import type { ImportPostSummary, InjectsImportInput } from '../../../../utils/api-types';
+import ImportUploaderInjectImportFile from './ImportUploaderInjectImportFile';
+import ImportUploaderInjectImportInjects from './ImportUploaderInjectImportInjects';
+import { InjectContext } from '../Context';
+import { storeXlsFile } from '../../../../actions/xls_formatter/xls-formatter-actions';
+
+const ImportUploaderInject = () => {
```
Comment (RomuDeuxfois) [inline openbas-front/src/admin/components/common/injects/ImportUploaderInject.tsx:12]: ImportUploaderInjectFromXls

File: openbas-front/src/admin/components/common/injects/ImportUploaderInjectImportFile.tsx [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+interface FormProps {
+  file: File;
+}
+
+interface Props {
+  handleClose: () => void;
+  handleSubmit: (values: FormProps) => void;
+}
+
+const ImportUploaderInjectImportFile: FunctionComponent<Props> = ({
```
Comment (RomuDeuxfois) [inline openbas-front/src/admin/components/common/injects/ImportUploaderInjectImportFile.tsx:35]: ImportUploaderInjectFromXlsFile

File: openbas-front/src/admin/components/common/injects/ImportUploaderInjectImportInjects.tsx [hunk 10]
```diff
@@ -0,0 +1,220 @@
...
+  timezone: string;
+}
+
+interface Props {
+  sheets: string[];
+  handleClose: () => void;
+  handleSubmit: (input: InjectsImportInput) => void;
+}
+
+const ImportUploaderInjectImportInjects: FunctionComponent<Props> = ({
```
Comment (RomuDeuxfois) [inline openbas-front/src/admin/components/common/injects/ImportUploaderInjectImportInjects.tsx:50]: ImportUploaderInjectFromXlsInjects

File: openbas-front/src/admin/components/settings/data_ingestion/xls_formatter/XlsMapperTestDialog.tsx [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+    >
+      <>
+        {importId === null
+          && <ImportUploaderInjectImportFile
+            handleClose={handleClose}
+            handleSubmit={onSubmitImportFile}
+             />
+        }
+        {importId !== null
+          && <ImportUploaderInjectImportInjectsTest
```
Comment (RomuDeuxfois) [inline openbas-front/src/admin/components/settings/data_ingestion/xls_formatter/XlsMapperTestDialog.tsx:55]: ImportUploaderInjectFromInjectsTest

File: openbas-api/src/main/java/io/openbas/rest/mapper/form/MapperAddInput.java (hunk 2 above)
Comment (Dimfacion) [reply openbas-api/src/main/java/io/openbas/rest/mapper/form/MapperAddInput.java:17]: Yes, I'm on it

**Review by Dimfacion — APPROVED** [review]

**Review by johanah29 — APPROVED** [review]

**Pull request merged**